storage/innobase/xtrabackup/src/backup_mysql.cc

```diff
@@ -2160,3 +2165,113 @@ mdl_unlock_all()
 	mutex_free(&mdl_lock_con_mutex);
 }
 
+bool
+has_innodb_buffer_pool_dump()
+{
+	if((server_flavor == FLAVOR_PERCONA_SERVER || 
+		server_flavor == FLAVOR_MYSQL) && mysql_server_version >= 50603) {
+		return true;
+	}
+		
+	if(server_flavor == FLAVOR_MARIADB && mysql_server_version >= 10000) {
+		return true;
+	}
+
+	return false;
+}
+
+bool
+has_innodb_buffer_pool_dump_pct()
+{
+	if((server_flavor == FLAVOR_PERCONA_SERVER || 
+		server_flavor == FLAVOR_MYSQL) && mysql_server_version >= 50702) {
+		return true;
+	}
+		
+	if(server_flavor == FLAVOR_MARIADB && mysql_server_version >= 10110) {
+		return true;
+	}
+
+	return false;
+}
+
+void
+dump_innodb_buffer_pool(MYSQL *connection)
+{
+	innodb_buffer_pool_dump_start_time = (ssize_t)my_time(MY_WME);
+
+	char *innodb_buffer_pool_dump_pct;
+	char *change_bp_dump_pct_query;
+
+	/* Verify if we need to change innodb_buffer_pool_dump_pct */
+	if(opt_dump_innodb_buffer_pool_pct != 0 && 
+		has_innodb_buffer_pool_dump_pct()) {
+		mysql_variable variables[] = {
+			{"innodb_buffer_pool_dump_pct", &innodb_buffer_pool_dump_pct},
+			{NULL, NULL}
+		};
+		read_mysql_variables(connection,
+			"SHOW GLOBAL VARIABLES LIKE 'innodb_buffer_pool_dump_pct'", 
+			variables, true);
+
+		original_innodb_buffer_pool_dump_pct = 
```

Comment:
@gl-sergei I'm not sure is ok to break the value assignment into 2 line, but leaving it on a single line would exceed 80 characters.